decidim-core/app/controllers/decidim/links_controller.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Decidim
+  class LinksController < Decidim::ApplicationController
+    helper Decidim::ExternalDomainHelper
+
```

Comment:
After we define the error, we can add a `rescue_from` here as follows:

```ruby
rescue_from Decidim::InvalidUrlError, with: :invalid_url
```

Then, define the `invalid_url` method in the controller:

```ruby
    private

    def invalid_url
      flash[:alert] = "Invalid URL" # add translation for this
      redirect_to decidim.root_path
    end
```